### PR TITLE
LC-2048-Implement forgot password journey-2

### PR DESCRIPTION
**Pull request by neeraj-jain-cbo**

### CODE REVIEW

File: src/main/java/uk/gov/cabinetoffice/csl/controller/reset/ResetFormValidator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        this.passwordPattern = passwordPattern;
+    }
+
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return ResetForm.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
```
Comment (SavvasCO) [inline src/main/java/uk/gov/cabinetoffice/csl/controller/reset/ResetFormValidator.java:25]: Maybe add a few tests for the `validate()` method in the test classes just to make sure they throw the correct errors where appropriate

**Review by SavvasCO — COMMENTED** [review]
Looks great. Just a small note.

**Comment by jaylco** [comment]
Nothing to add, looking good

**Review by jaylco — DISMISSED** [review]

File: src/main/java/uk/gov/cabinetoffice/csl/controller/reset/ResetFormValidator.java (hunk 1 above)
Comment (neeraj-jain-cbo) [reply src/main/java/uk/gov/cabinetoffice/csl/controller/reset/ResetFormValidator.java:25]: Following unit tests are added in the "ResetControllerTest" to validate the password:
shouldShowPasswordInvalidErrorMessageIfNumberIsMissingFromPassword
shouldShowPasswordInvalidErrorMessageIfLowerCaseCharacterMissingFromPassword
shouldShowPasswordInvalidErrorMessageIfUpperCaseCharacterMissingFromPassword
shouldShowPasswordInvalidErrorMessageIfPasswordLengthIsLessThan8Characters
shouldShowPasswordInvalidErrorMessageIfPasswordHasSpace
shouldShowPasswordMismatchErrorMessageIfPasswordAndConfirmPasswordMismatch

**Review by SavvasCO — DISMISSED** [review]
LGTM! 👍

**Review by SavvasCO — APPROVED** [review]

**Pull request merged**